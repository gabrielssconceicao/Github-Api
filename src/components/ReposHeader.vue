<script setup lang="ts">
const props = defineProps<{ name: string }>()
</script>

<template>
  <header class="bg-header p-2 flex items-center text-white">
    <RouterLink to="/" class="w-14 border-white border-2 p-1  flex justify-center items-center">Back
    </RouterLink>
    <p class="flex-1 text-center">
      <span class="font-bold font-mono text-xl text-primary">{{ props.name }}</span>
      Repos
    </p>
  </header>
</template>
