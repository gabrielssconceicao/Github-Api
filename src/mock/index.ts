import { type IUser } from '@/types/user';
import { type IRepos } from '@/types/repos';

export const user: IUser = {
  name: 'Teste User',
  login: 'testeuser',
  avatar_url: 'https://avatars.githubusercontent.com/u/10590076?v=4',
  followers: 0,
  following: 5,
  location: 'Brazil',
  html_url: 'https://github.com/testeuser',
  bio: 'lorem ipsum dolor sit amet',
  public_repos: 5,
};
// JS - 3
// HTML - 2
// python - 1
// Ruby - 1
//Java - 1
//C++ - 1
//Typescript - 1
export const repos: IRepos[] = [
  {
    name: 'Teste Repo 1',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 2',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'HTML',
  },
  {
    name: 'Teste Repo 3',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Python',
  },
  {
    name: 'Teste Repo 4',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 5',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 6',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'HTML',
  },
  {
    name: 'Teste Repo 7',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Ruby',
  },
  {
    name: 'Teste Repo 8',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Java',
  },
  {
    name: 'Teste Repo 9',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'C++',
  },
  {
    name: 'Teste Repo 10',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Typescript',
  },
];
