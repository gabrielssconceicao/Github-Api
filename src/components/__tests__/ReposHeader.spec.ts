import { describe, expect, it } from 'vitest';
import { mount } from '@vue/test-utils';
import ReposHeader from '../ReposHeader.vue';
import { user } from '@/mock';

describe('<ReposHeader />', () => {
  it('should render with userName', () => {
    const wrapper = mount(ReposHeader, {
      props: {
        name: user.name as string,
      },
    });

    expect(wrapper.find('header').exists()).toBe(true);
    expect(wrapper.find('p').text()).toBe(`${user.name} Repos`);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render with login', () => {
    const wrapper = mount(ReposHeader, {
      props: {
        name: null || user.login,
      },
    });
    expect(wrapper.find('header').exists()).toBe(true);
    expect(wrapper.find('p').text()).toBe(`${user.login} Repos`);
    expect(wrapper.element).toMatchSnapshot();
  });
});
