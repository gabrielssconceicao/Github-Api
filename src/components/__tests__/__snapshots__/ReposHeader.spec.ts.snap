// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ReposHeader /> > should render with login 1`] = `
<header
  class="bg-header p-2 flex items-center text-white"
>
  <routerlink
    class="w-14 border-white border-2 p-1 flex justify-center items-center"
    to="/"
  >
    Back 
  </routerlink>
  <p
    class="flex-1 text-center"
  >
    <span
      class="font-bold font-mono text-xl text-primary"
    >
      testeuser
    </span>
     Repos 
  </p>
</header>
`;

exports[`<ReposHeader /> > should render with userName 1`] = `
<header
  class="bg-header p-2 flex items-center text-white"
>
  <routerlink
    class="w-14 border-white border-2 p-1 flex justify-center items-center"
    to="/"
  >
    Back 
  </routerlink>
  <p
    class="flex-1 text-center"
  >
    <span
      class="font-bold font-mono text-xl text-primary"
    >
      Teste User
    </span>
     Repos 
  </p>
</header>
`;
